core.py

```diff
@@ -358,8 +358,8 @@ def Yield_r(r): return Efield_r(r) / (units.kV / units.cm) - \
         uniform_to_emission_time = interp1d(np.cumsum(dy), np.cumsum(dt),
                                             bounds_error=False, fill_value=(0, sum(dt)))
 
-        probabilities = 1 - np.random.uniform(0, 1, size=shape)
-        return uniform_to_emission_time(probabilities)
+        reshape_Y = np.linspace(0, 1, 1001); reshape_X = uniform_to_emission_time(reshape_Y)
+        return reshape_X[np.around(1 - np.random.uniform(0, 1, size=shape)/1000).astype(int)]
```

Comment:
Reduce interpolation time by using evenly spaced out points, 1000 samples corresponds to ~0.35 ns time resolution.